File: principles-and-acronyms/solid/single_responsability_principle/nodejs/tarea01.js
// Clase MovieProjectorOperator
var MovieProjectorOperator = /** @class */ (function () {
    function MovieProjectorOperator() {
    }
    MovieProjectorOperator.prototype.startProjection = function () {
        console.log("Starting projection ...");
    };
    MovieProjectorOperator.prototype.stopProjection = function () {
        console.log("Stopping projection ...");
    };
    MovieProjectorOperator.prototype.checkProjectorStatus = function () {
        return "Projector is running";
    };
    return MovieProjectorOperator;
}());
// Clase TicketSellerEmployee
var TicketSellerEmployee = /** @class */ (function () {
    function TicketSellerEmployee() {
    }
    TicketSellerEmployee.prototype.sellTicket = function () {
        console.log("Selling ticket ...");
    };
    TicketSellerEmployee.prototype.refundTicket = function () {
        console.log("Refunding ticket ...");
    };
    TicketSellerEmployee.prototype.checkTicketAvailability = function () {
        return 100;
    };
    return TicketSellerEmployee;
}());
// Clase ConcessionStandStaff
var ConcessionStandStaff = /** @class */ (function () {
    function ConcessionStandStaff() {
    }
    ConcessionStandStaff.prototype.serveSnack = function () {
        console.log("Serving snack ...");
    };
    ConcessionStandStaff.prototype.restockItems = function () {
        console.log("Restocking items ...");
    };
    ConcessionStandStaff.prototype.processPayment = function () {
        console.log("Processing payment ...");
    };
    return ConcessionStandStaff;
}());
// Clase principal del sistema de gestión de cine
var Tarea01 = /** @class */ (function () {
    function Tarea01() {
    }
    Tarea01.main = function (args) {
        var movieProjectorOperator = new MovieProjectorOperator();
        var ticketSellerEmployee = new TicketSellerEmployee();
        var concessionStandStaff = new ConcessionStandStaff();
        movieProjectorOperator.startProjection();
        ticketSellerEmployee.sellTicket();
        concessionStandStaff.serveSnack();
    };
    return Tarea01;
}());
Tarea01.main([]);

File: principles-and-acronyms/solid/single_responsability_principle/java/tarea01.java
// Interfaz MovieProjector
interface MovieProjector {
    void startProjection();
    void stopProjection();
    String checkProjectorStatus();
}

// Interfaz TicketSeller
interface TicketSeller {
    void sellTicket();
    void refundTicket();
    int checkTicketAvailability();
}

// Interfaz ConcessionStandWorker
interface ConcessionStandWorker {
    void serveSnack();
    void restockItems();
    void processPayment();
}

// Clase MovieProjectorOperator
class MovieProjectorOperator implements MovieProjector {
    public void startProjection() {
        System.err.println("Starting projection ...");
    }

    public void stopProjection() {
        System.err.println("Stopping projection ...");
    }

    public String checkProjectorStatus() {
        return "Projector is running";
    }
}

// Clase TicketSellerEmployee
class TicketSellerEmployee implements TicketSeller {
    public void sellTicket() {
        System.err.println("Selling ticket ...");
    }

    public void refundTicket() {
        System.err.println("Refunding ticket ...");
    }

    public int checkTicketAvailability() {
        return 100;
    }
}

// Clase ConcessionStandStaff
class ConcessionStandStaff implements ConcessionStandWorker {
    public void serveSnack() {
        System.err.println("Serving snack ...");
    }

    public void restockItems() {
        System.err.println("Restocking items ...");
    }

    public void processPayment() {
        System.err.println("Processing payment ...");
    }
}

// Clase principal del sistema de gestión de cine
public class tarea01 {
    public static void main(String[] args) {
        MovieProjectorOperator movieProjectorOperator = new MovieProjectorOperator();
        TicketSellerEmployee ticketSellerEmployee = new TicketSellerEmployee();
        ConcessionStandStaff concessionStandStaff = new ConcessionStandStaff();

        movieProjectorOperator.startProjection();
        ticketSellerEmployee.sellTicket();
        concessionStandStaff.serveSnack();
    }
}
